### PR TITLE
🚧Implement Carbonate library

### LINKED ISSUE #473: 🚧Implement Carbonate library
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Add the library **_Carbonate_** and refactor the code as required to integrate and replace the `Reactable` related code with _**Carbonate**_ code.

### Acceptance Criteria

- [x] _**Carbonate**_ library/NuGet package added to the required projects
- [x] Old `Reactable` code refactored and replaced with _**Carbonate**_ code.

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [X] Issue linked to the correct project _(if applicable)_.
- [X] Issue linked to the correct milestone _(if applicable)_.
- [x] Draft pull request created and linked to this issue _(only required with code changes)_.

### Issue Dependencies

_No response_

### Related Work

_No response_

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| Workflow Changes    | `workflow` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/feature/<issue-num>-<description>
      Example: preview/feature/123-my-preview-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_preview release_** branch, with the correct naming syntax, in the following example:
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
    Example: preview/v1.2.3-preview.4
    ```
  </details>

- [x] The **_base(target)_** branch for this pull request is a **_preview release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

💡For more information on branching, refer to the project's [branching documentation](../../Documentation/Branching.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title matches the title of the associated issue.

- [x] A `preview` label has been added to the pull request.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out, and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

**Comment by CalvinWilkinson** [comment]
Put https://github.com/KinsonDigital/Velaptor/labels/on%20hold

**Reason:** refer to linked issue [comment](https://github.com/KinsonDigital/Velaptor/issues/473#issuecomment-1361557296) for more information.

File: Velaptor/ReactableData/BatchSizeData.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿// <copyright file="BatchSizeData.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+namespace Velaptor.ReactableData;
+
+using Carbonate;
+
+/// <summary>
+/// Holds data for the <see cref="Reactable"/> reactable.
```
Comment (KinsonDigitalAdmin) [inline Velaptor/ReactableData/BatchSizeData.cs:10]: Refactor to:

```suggestion
/// Holds data for the the size of the batch.
```

File: Velaptor/Content/Caching/SoundCache.cs [hunk 2]
```diff
@@ -38,39 +38,32 @@ internal sealed class SoundCache : IItemCache<string, ISound>
...
+        this.shutDownUnsubscriber = this.reactable.Subscribe(new Reactor(
+            eventId: NotificationIds.ShutDownId,
+            onNext: ShutDown));
     }
 
     /// <summary>
     /// Finalizes an instance of the <see cref="SoundCache"/> class.
     /// </summary>
-    [ExcludeFromCodeCoverage]
+    [ExcludeFromCodeCoverage(Justification = "De-constructors cannot be unit tested.")]
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Content/Caching/SoundCache.cs:66]: Add justifications to all of the `[ExcludeFromCodeCoverage]` attributes.

File: Velaptor/NotificationIds.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    /// <summary>
+    /// Gets the unique event <see cref="Guid"/> for push notifications to dispose of sounds.
+    /// </summary>
+    public static Guid DisposeSoundId { get; } = new ("863983d2-6657-4c8e-8e9a-f3cbd688abe1");
+
+    /// <summary>
+    /// Gets the unique event <see cref="Guid"/> for push notifications to shut down.
+    /// </summary>
+    public static Guid ShutDownId { get; } = new ("17b9fd1c-67ef-4f36-8973-45e32b0ee85b");
```
Comment (KinsonDigitalAdmin) [inline Velaptor/NotificationIds.cs:53]: Give these property names better past-tense type event names.

Example: `GLInitId `should be changed to `GLInitializedId`

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

File: Velaptor/ReactableData/KeyboardKeyStateData.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+/// </summary>
+public sealed record KeyboardKeyStateData
+{
+    /// <summary>
+    /// Gets or sets the key code.
+    /// </summary>
+    public KeyCode Key { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the current <see cref="Key"/> is in the down position.
```
Comment (kselena) [inline Velaptor/ReactableData/KeyboardKeyStateData.cs:20]: Grammar
```suggestion
    /// Gets or sets a value indicating whether or not the current <see cref="Key"/> is in the down position.
```

**Review by kselena — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Review by kselena — APPROVED** [review]
